### PR TITLE
keep active stats lock held

### PR DESCRIPTION
**Pull request by jeffwashington**
#### Problem
`let _ = ...` causes rhs to be dropped immediately.
So, active period was 0s.

#### Summary of Changes
use
`let _guard = ...`

Fixes #

### CODE REVIEW

File: runtime/src/read_only_accounts_cache.rs [hunk 1]
```diff
@@ -10,7 +10,7 @@ use {
         pubkey::Pubkey,
     },
     std::sync::{
-        atomic::{AtomicU64, AtomicUsize, Ordering},
+        atomic::{AtomicU32, AtomicU64, AtomicUsize, Ordering},
```
Comment (HaoranYi) [inline runtime/src/read_only_accounts_cache.rs:13]: should this be in a different PR?

**Comment by HaoranYi** [comment]
Should the `ReadOnlyAccountCacheEntry` be in a different PR?

**Comment by jeffwashington** [comment]
> Should the `ReadOnlyAccountCacheEntry` change be in a different PR?

yeah, these were cherry picked from master so I could run diagnostics and test these changes out when master was broken recently against mnb. That is fixed now. Rebasing to master got rid of those changes.

**Review by HaoranYi — APPROVED** [review]
lgtm
